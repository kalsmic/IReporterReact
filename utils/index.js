export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const baseURL = 'https://ireporterv3.herokuapp.com/api/v3';

export const isAuthenticated = () => {
  const token = sessionStorage.getItem('iReporterToken');
  if (token) { return true; }
  return false;
};
