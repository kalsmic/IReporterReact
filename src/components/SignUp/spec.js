import { shallow } from 'enzyme';
import React from 'react';
import { SignUp, mapStateToProps, mapDispatchToProps } from './index';
import { findByTestAttr } from '../../../utils';

describe('Sign up Component', () => {
  const signUpUserFn = jest.fn();
  const props = {
    signUpUser: signUpUserFn,
    message: '',
    isLoading: false,
    error: {},

  };
  const newUserData = [
    ['username', 'kalsmic'],
    ['email', 'admin@eemail.com'],
    ['password', 'Pa$$word123'],
  ];
  const wrapper = shallow(<SignUp {...props} />);

  it('should render without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the onchange event', () => {
    const mockHandleChange = (name, value) => {
      const event = {
        target: {
          name,
          value,
        },
      };

      wrapper.instance().handleChange(event);
    };

    // eslint-disable-next-line array-callback-return
    newUserData.map(([name, value]) => {
      mockHandleChange(name, value);
      expect(wrapper.instance().state[name]).toBe(value);
    });
  });

  it('should handle the onSubmit event', () => {
    wrapper.setState({
      username: '',
      email: '',
      password: ''
    });
    const instance = wrapper.instance();
    const event = {
      target: {
        type: 'submit',
        name: 'login',
      },
      preventDefault: jest.fn()
    };
    instance.handleSubmit(event);
    expect(instance.state.username).toBe('');
    expect(instance.state.password).toBe('');
    expect(instance.props.signUpUser).not.toBeCalled();

    newUserData.map(([name, value]) => (
      wrapper.setState({
        [name]: value
      })
    ));
    instance.handleSubmit(event);

    expect(instance.props.signUpUser).toBeCalled();
  });

  it('should show messages to the user', () => {
    const error = {
      username: 'username Errors',
      email: 'email Error',
      password: 'password Error'
    };
    wrapper.setProps({
      error,
      isLoading: true,
      message: 'Success message'
    });
    expect(findByTestAttr(wrapper, 'usernameError').text()).toBe(error.username);
    expect(findByTestAttr(wrapper, 'emailError').text()).toBe(error.email);
    expect(findByTestAttr(wrapper, 'passwordError').text()).toBe(error.password);
    expect(findByTestAttr(wrapper, 'loader').exists()).toBe(true);
    expect(findByTestAttr(wrapper, 'message').text()).toBe('Success message');
  });


  it('should map state to props', () => {
    const mockedState = {
      signUpReducer: {
        message: 'Account created Successfully',
        isLoading: false,
        error: ''
      }
    };
    const state = mapStateToProps(mockedState);
    expect(state.message).toBe('Account created Successfully');
    expect(state.isLoading).toBe(false);
    expect(state.error).toBe('');
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch).signUpUser();
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
