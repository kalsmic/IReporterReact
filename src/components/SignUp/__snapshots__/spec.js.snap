// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Component should render without crushing 1`] = `
<div
  className="wrapper"
>
  <Header />
  <div
    className="wrapper__login_register border-round-lg"
  >
    <h3>
      Create an account !
    </h3>
    <form
      className="wrapper__login_register__form"
      onSubmit={[Function]}
    >
      <input
        className="wrapper__login_register__form__input"
        name="username"
        onChange={[Function]}
        placeholder="USERNAME"
        required={true}
        type="text"
        value=""
      />
      <input
        className="wrapper__login_register__form__input"
        name="email"
        onChange={[Function]}
        placeholder="EMAIL ADDRESS"
        required={true}
        type="email"
        value=""
      />
      <input
        className="wrapper__login_register__form__input"
        name="password"
        onChange={[Function]}
        placeholder="PASSWORD"
        required={true}
        type="password"
        value=""
      />
      <button
        className="wrapper__login_register__form__submit"
        name="signup"
        type="submit"
      >
        SIGNUP
      </button>
      <Link
        replace={false}
        to="/"
      >
         Click Here to Login
      </Link>
    </form>
  </div>
  <Footer />
</div>
`;
